pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/RenderingServlet.java

```diff
@@ -0,0 +1,77 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.asciidoctor.AsciidoctorService;
+import com.redhat.pantheon.model.Rendering;
+import com.redhat.pantheon.model.module.HashableFileResource;
+import com.redhat.pantheon.model.module.Module;
+import com.redhat.pantheon.model.module.ModuleLocale;
+import com.redhat.pantheon.model.module.SourceContent;
+import org.apache.commons.lang3.LocaleUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.redhat.pantheon.conf.GlobalConfig.DEFAULT_MODULE_LOCALE;
+import static com.redhat.pantheon.model.module.ModuleVariant.DEFAULT_VARIANT_NAME;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValue;
+import static com.redhat.pantheon.servlet.ServletUtils.paramValueAsBoolean;
+
+/**
+ * Renders an HTML preview for a single module.
+ * To provide parameters to the asciidoc generation process, provide the parameters with their name prefixed
+ * with "ctx_".
+ *
+ * For example, if an asciidoc attribute of name 'product' needs to be passed, there will need to be a
+ * query parameter of name 'ctx_product' provided in the url.
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet which transforms asciidoc content into html",
+                Constants.SERVICE_VENDOR + "=Red Hat Content Tooling team"
+        })
+@SlingServletResourceTypes(
+        resourceTypes = { "pantheon/module" },
```

Comment:
You are missing the `pantheon/assembly` resource type here if you want to use the same servlet to preview both.

Reply:
Are we sure we want to use the same servlet for both?

In @xdavidson's recent PR, we agreed that our gut was that the upload servlets would likely be sufficiently different to warrant them being separate. I wonder if the same doesn't hold true for the rendering servlets?

I admit that off the top of my head, I'm not sure why the two rendering servlets might look different, at least from a client perspective, especially since this is for single-page assemblies. But I'm interested to hear other opinions on whether we should have one render servlet or two separate.

Reply:
I have tried to create separate rendering servlet, to resolve this issue, considering, we have different resourcetype  as well for Module and assembly

Reply:
Ok. I think that works for now. We can revisit the decision later.

Reply:
(if necessary)